core/PruManager.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * PruManager.cpp
+ *
+ *	Created on: July 3, 2021
+ *		Author: Dhruva Gole
+ */
+
+#include "../include/PruManager.h"
+#include "MiscUtilities.h"
+#include <iostream>
+
+#if ENABLE_PRU_RPROC == 1
+PruManagerRprocMmap::PruManagerRprocMmap(unsigned int pruNum, unsigned int v)
+{
+	/* constructor for initializing the necessary path variables
+	 * based on the value of pru_num to choose:
+	 * 0 for PRU1 core 0
+	 * 1 for PRU1 core 1
+	 * 2 for PRU2 core 0
+	 * 3 for PRU2 core 1
+	 */
+	pru_num = pruNum;
+	verbose = v;
+	unsigned int pruss = pru_num / 2 + 1;
+	unsigned int prucore = pru_num % 2;
+
+	basePath = "/dev/remoteproc/pruss" + std::to_string(pruss) + "-core" + std::to_string(prucore) + "/";
+	statePath = basePath + "state";
+	firmwarePath = basePath + "firmware";
+	firmware = "am57xx-pru" + std::to_string(pruss) + "_" + std::to_string(prucore) + "-fw";
+# ifdef firmwareBelaRProc
+	std::string firmwareBela = firmwareBelaRProc;	// Incoming string from Makefile
+# endif	// firmwareBelaRProc
```

Comment:
as written, if `firmwareBelaRProc` is not defined, you will be left  with a confusing compiler error at the next line. If the definition of `firmwareBelaRProc` is required, then add an explicit
```
#else // firmwareBelaRProc
#error (meaningful error description here)
#endif // firmwareBelaRProc
```